### PR TITLE
Use the K8s SIG iptables wrapper

### PR DESCRIPTION
**Pull request by skitt**
... instead of our own. This ensures that our container images use the
correct iptables implementation without placing any particular
requirements on the host. It also removes the requirement for mounting
the host file system in the container (which appears to be restricted
in OpenShift 4.7 and later).

This requires installing both variants of iptables in the container;
we can do that on Fedora, but not on UBI8, so this is limited to the
gateway image for the time being.

See https://github.com/kubernetes-sigs/iptables-wrappers for details.

Signed-off-by: Stephen Kitt <skitt@redhat.com>

### CODE REVIEW

**Comment by mangelajo** [comment]
@nyechiel this looks more like a change for 0.9, can we move it to that board instead?

**Comment by nyechiel** [comment]
> @nyechiel this looks more like a change for 0.9, can we move it to that board instead?

Done

**Comment by skitt** [comment]
This is still relevant.

File: package/submariner.sh [hunk 1]
```diff
@@ -18,34 +18,6 @@ mkdir -p /etc/ipsec
...
-    echo "$f is present on the host at ${location}/$f"
-	sed "s!@@PATH@@!${location}!" /usr/sbin/iptables-wrapper.in > /usr/sbin/$f
-  else
-    echo "WARNING: not using iptables wrapper because iptables was not detected on the"
-    echo "host at the following paths [/usr/sbin, /sbin]."
-    echo "Either the host file system isn't mounted or the host does not have iptables"
-    echo "installed. The pod will use the image installed iptables version."
-  fi
-done
+export PATH=$PATH:/usr/libexec/strongswan
```
Comment (mangelajo) [inline package/submariner.sh]: We don't need that right ^ no more strongswan

**Review by mangelajo — CHANGES_REQUESTED** [review]
the path for strongswan isn't necessary anymore

File: package/submariner.sh (hunk 1 above)
Comment (skitt) [reply package/submariner.sh]: Aargh yes I meant to delete that

**Review by mangelajo — APPROVED** [review]

**Review by sridhargaddam — APPROVED** [review]

**Comment by nyechiel** [comment]
@skitt @mangelajo this seems critical for 0.9, but should we also consider backport to 0.8?

**Comment by skitt** [comment]
> @skitt @mangelajo this seems critical for 0.9, but should we also consider backport to 0.8?

Once we figure out whether it actually is required for OCP 4.7, yes. If it _is_ required, we’d also need to apply this to the other `iptables`-using images and backport that as well.

**Pull request merged**